handsontable/src/plugins/filters/ui/multipleSelect.js

```diff
@@ -60,10 +60,18 @@ export class MultipleSelectUI extends BaseUI {
    * @type {LinkUI}
    */
   #clearAllUI;
+  /**
+   * Whether to uncheck filtered queries.
+   *
+   * @type {boolean}
+   */
+  uncheckFilteredQueries = false;
 
   constructor(hotInstance, options) {
     super(hotInstance, extend(MultipleSelectUI.DEFAULTS, options));
 
+    this.uncheckFilteredQueries = options.uncheckFilteredQueries;
```

Comment:
All options passed in the constructor are available as `this.options. uncheckFilteredQueries`. There is no need to add a custom field here.